ros2topic/test/test_bw_delay_hz.py

```diff
@@ -145,6 +145,15 @@ def publish_message():
 
                 publish_timer = self.node.create_timer(0.5, publish_message)
 
+                # Wait for the publisher to be discovered
+                publisher_count = 0
+                timeout_count = 0
+                while publisher_count == 0 and timeout_count < 10:
+                    self.executor.spin_once(timeout_sec=0.1)
+                    publisher_count = self.node.count_publishers(topic)
+                    timeout_count += 1
+                assert publisher_count > 0, 'Publisher was not discovered'
```

Comment:
we can confirm here that the publisher is online before any test commands are executed below.